sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/KinesisIO.java

```diff
@@ -420,6 +426,47 @@ public Read withCustomWatermarkPolicy(WatermarkPolicyFactory watermarkPolicyFact
       return toBuilder().setWatermarkPolicyFactory(watermarkPolicyFactory).build();
     }
 
+    /**
+     * Specifies the rate limit policy as BackoffRateLimiter.
+     *
+     * @param fluentBackoff The {@code FluentBackoff} used to create the backoff policy.
+     */
+    public Read withBackoffRateLimitPolicy(FluentBackoff fluentBackoff) {
```

Comment:
Please, add an example into KinesisIO javadoc how to use it.

Reply:
org.apache.beam.sdk.util.FluentBackoff is not meant to be on the public API surface and is only meant for internal use. All classes within org.apache.beam.sdk.util are like that.

If we would want to promote it to be public, we should find the appropriate place for it and move it out of util.

Reply:
What's about wrapping it into KinesisIO specific backoff wrapper? 
PS: it's not evident that this is for internal use only. Do we need to annotate it?

Reply:
@lukecwik In this case, I suggest to change a signature of this method to `Read withBackoffRateLimitPolicy(int maxRetries, Duration maxCumulativeBackoff)` and insatiate `FluentBackoff` internally with provided params. 
PS: Updated `sdk/util package-info.java` doc,  #10009

Reply:
I was considering exposing a wrapper object like SnsIO. I didn't realize FluentBackoff was internal. I think that makes a lot of sense here too then.

What does it mean to give up retrying with an unbounded source though? What should we do in that case? Throw an unchecked exception and let the pipeline crash? That's why I explicitly overrode max retries and max cumulative backoff because I didn't think that made sense but I can change that if you want.

Reply:
@jfarr I'd leave this responsibility to user and that's why I proposed to expose these parameters.